packages/core/src/components/forms/fileInput.tsx

```diff
@@ -108,11 +128,25 @@ export class FileInput extends React.PureComponent<IFileInputProps, {}> {
         return (
             <label {...htmlProps} className={rootClasses}>
                 <input {...inputProps} onChange={this.handleInputChange} type="file" disabled={disabled} />
-                <span className={Classes.FILE_UPLOAD_INPUT}>{text}</span>
+                <span ref={this.createRef} className={Classes.FILE_UPLOAD_INPUT}>
+                    {text}
+                </span>
             </label>
         );
     }
 
+    private createRef = (input: HTMLSpanElement | null) => {
+        if (input !== null) {
+            this.input = input;
+        }
+    };
+
+    private changeButtonText = (text: string) => {
+        if (this.input) {
+            this.input.setAttribute("button-text", text);
```

Comment:
we should namespace the attribute name, just to be safe
```suggestion
            this.input.setAttribute(`${NS}-button-text`, text);
```

... this would also require exporting `NS` from `packages/core/src/common/classes.ts` and importing that constant here